src/Phan/Plugin/Internal/DependentReturnTypeOverridePlugin.php

```
@@ -93,6 +93,38 @@ public static function getReturnTypeOverridesStatic(CodeBase $code_base) : array
                 }
             };
         };
+        /**
+         * @phan-return Closure(CodeBase,Context,Func,array):UnionType
+         */
+        $make_nonzero_dependent_type_method = static function (int $argument_pos, UnionType $type_if_zero, UnionType $type_if_nonzero, UnionType $type_if_unknown) : Closure {
+            /**
+             * @param Func $function @phan-unused-param
+             * @param list<Node|int|float|string> $args
+             */
+            return static function (
+                CodeBase $code_base,
+                Context $context,
+                Func $function,
+                array $args
+            ) use (
+                $type_if_zero,
+                $type_if_unknown,
+                $type_if_nonzero,
+                $argument_pos
+            ) : UnionType {
+                if (count($args) <= $argument_pos) {
+                    return $type_if_unknown;
+                }
+                $result = (new ContextNode($code_base, $context, $args[$argument_pos]))->getEquivalentPHPScalarValue();
+                if (is_numeric($result)) {
+                    if ($result == 0) {
+                        return $type_if_zero;
```

I don't expect the factory $make_nonzero_dependent_type_method to be reused elsewhere due to other functions not having the same edge cases - just directly add a closure for $bcdiv_callback = function() {}.

PHP 8 will throw a DivisionByZero error instead of returning null - src/Phan/Language/Internal/FunctionSignatureMap_php80_delta.php should be updated for this

```
php > var_export(bcdiv(1,0));

Warning: Uncaught DivisionByZeroError: Division by zero in php shell code:1
Stack trace:
#0 php shell code(1): bcdiv('1', '0')
#1 {main}
  thrown in php shell code on line 1
php > echo PHP_VERSION;
8.0.0-dev

php > var_export(bcmod(1,0));

Warning: Uncaught DivisionByZeroError: Modulo by zero in php shell code:1
Stack trace:
#0 php shell code(1): bcmod('1', '0')
#1 {main}
  thrown in php shell code on line 1

```

So this would be

```
if (Config::getValue('target_php_version') >= 80000) {
    return $type_if_nonzero;
}
```

---

Thank you for your feedback, will fix.

---

My last commit adresses all your suggestions. Thanks again!